File: Interview150/383_ransomNote.js
var canConstruct = function(ransomNote, magazine) {
    let mpp = new Map();
    for(let char of magazine){
        mpp.set(char, (mpp.get(char) || 0)+1);
    }
    for(let char of ransomNote){
        if(!mpp.has(char) || mpp.get(char) === 0) return false;
        mpp.set(char, mpp.get(char) - 1);
    }
    return true;
};

File: C++/Arrays/383_ransomNote.cpp
#include<bits/stdc++.h>
using namespace std;

bool canConstruct(string ransom, string magazine){
  if(ransom.size() > magazine.size()) return false;
  unordered_map<char, int> mpp;
  for(auto& i : magazine) mpp[i]++;

  for(auto& i : ransom){
    if(mpp[i] <= 0) return false;
    mpp[i]--;
  }
  return true;
}
